docker/proof-server-chart/templates/proof-server-deployment.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "proof-server.fullname" . }}-deployment
+  labels:
+  {{- include "proof-server.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.deployment.replicas }}
+  selector:
+    matchLabels:
+      app: proof-server
+    {{- include "proof-server.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app: proof-server
+      {{- include "proof-server.selectorLabels" . | nindent 8 }}
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.deployment.proofServer.terminationGracePeriodSeconds }}
+      nodeSelector:
+        {{- toYaml .Values.deployment.proofServer.nodeSelector | nindent 8 }}
+      containers:
+      - command:
+        - sh
+        - -c
+        - /app/proof_server --mode "single" --addr ${CONTAINER_ADDR}:${CONTAINER_PORT}
+        env:
+        - name: RUST_LOG
+          value: {{ quote .Values.deployment.proofServer.env.rustLog }}
+        - name: RUST_MIN_STACK
+          value: {{ quote .Values.deployment.proofServer.env.rustMinStack }}
+        - name: SHARD_SIZE
+          value: {{ quote .Values.deployment.proofServer.env.shardSize }}
+        - name: SHARD_BATCH_SIZE
+          value: {{ quote .Values.deployment.proofServer.env.shardBatchSize }}
+        - name: SHARD_CHUNKING_MULTIPLIER
+          value: {{ quote .Values.deployment.proofServer.env.shardChunkingMultiplier }}
+        - name: RECONSTRUCT_COMMITMENTS
+          value: {{ quote .Values.deployment.proofServer.env.reconstructCommitments }}
+        - name: CONTAINER_ADDR
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: CONTAINER_PORT
+          value: {{ quote .Values.deployment.proofServer.env.containerPort }}
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ quote .Values.kubernetesClusterDomain }}
+        image: {{ .Values.deployment.proofServer.image.repository }}:{{ .Values.deployment.proofServer.image.tag
+          | default .Chart.AppVersion }}
+        imagePullPolicy: {{ .Values.deployment.proofServer.imagePullPolicy }}
+        livenessProbe:
+          httpGet:
+            path: /health
```

Comment:
Do we want to put the path of the health/readiness probes configurable via `values.yaml`?